test/unit/kubernetes-deploy/kubernetes_resource_test.rb

```diff
@@ -288,6 +288,17 @@ def test_disappeared_is_false_if_resource_has_been_deployed_and_we_get_a_server_
     refute_predicate(dummy, :disappeared?)
   end
 
+  def test_custom_resource_kind
```

Comment:
Can you make this name more descriptive and/or add an `assert_nothing_raised` and/or assert that `build` returned an instance of the expected class? I'm thinking it won't be very obvious in the future what this test is covering / what the bug was.